### ISSUE TITLE
Broken indentation inside with statements

### ISSUE THREAD

**Issue by voger** [elixir-editors/vim-elixir#385]
For some reason I get weird indentation inside with statements. Here is an example
# Actual

```ex
  def resend_confirmation(username) when is_binary(username) do
    with user = %User{confirmed_at: nil} <- get_by(username: username) do
      {:ok, user} = 
        user 
        |> DB.add_confirm_token 
        |> update_user() 
                                                                         Log.info(%Log{user: user.id, message: "send new confirmation"}) 
                                                                         send_welcome(user)
                                                                         {:ok, user}
    else
      nil ->
        {:error, "not found"}
      %User{email: email} ->
        Email.already_confirmed(email)
        {:error, "already confirmed"}
    end
  end

```

# Expected

```ex
  def resend_confirmation(username) when is_binary(username) do
    with user = %User{confirmed_at: nil} <- get_by(username: username) do
      {:ok, user} = 
        user 
        |> DB.add_confirm_token 
        |> update_user() 
        Log.info(%Log{user: user.id, message: "send new confirmation"}) 
        send_welcome(user)
        {:ok, user}
    else
      nil ->
        {:error, "not found"}
      %User{email: email} ->
        Email.already_confirmed(email)
        {:error, "already confirmed"}
    end
  end

```

Another example

# Actual

```ex
  def create_user(params) do
    profile = UserProfile.registration_changeset(%UserProfile{}, params)

    user_cs =
      %User{}
      |> User.registration_changeset(params)
      |> put_assoc(:user_profile, profile)

    with {:ok, user} <- Repo.insert(user_cs, returning: false) do
      Log.info(%Log{user: user.id, message: "user created"})
                                                                 send_welcome(user)
                                                                 {:ok, user}
    end
  end
```

# Expected

```ex
  def create_user(params) do
    profile = UserProfile.registration_changeset(%UserProfile{}, params)

    user_cs =
      %User{}
      |> User.registration_changeset(params)
      |> put_assoc(:user_profile, profile)

    with {:ok, user} <- Repo.insert(user_cs, returning: false) do
      Log.info(%Log{user: user.id, message: "user created"})
      send_welcome(user)
      {:ok, user} 
    end
  end
```

P.S.: Is there any way to disable indentation temporarily until a solution for this issue is found?

**Comment by jbodah** [comment]
Thanks for this. `with` has a few known issues but it's good to have these for test cases

> P.S.: Is there any way to disable indentation temporarily until a solution for this issue is found?

I would recommend forking this repo and stubbing this function to just `return -1` https://github.com/elixir-editors/vim-elixir/blob/master/autoload/elixir/indent.vim#L319

**Comment by dylan-chong** [comment]
This situation may also be relevant

When `with` is in a sigil, it seems to be able to confuse the indenting

```
  r = ~r"with"
              waaaaaaa

```

If `~r` is removed, indenting works as expected

**Comment by clarkenciel** [comment]
having `with` in comments also seems to break indentation after the comment.

```elixir
# This file is responsible for configuring your application
# and its dependencies with the aid of the Mix.Config module.
                            use Mix.Config

                            import_config "#{Mix.env}.exs"
```

vs.

```elixir
# This file is responsible for configuring your application
# and its dependencies 
use Mix.Config

import_config "#{Mix.env}.exs"
```

**Comment by jbodah** [comment]
@clarkenciel thanks for the report, pulling this out into it's own issue

**Comment by jbodah** [comment]
Fixed in 1543e37f5940396b3cd18f67f1c36530d289314c
